sokutei.h

```diff
@@ -0,0 +1,188 @@
+#ifndef SOKUTEI_BENCHMARK_H
+#define SOKUTEI_BENCHMARK_H
+
+#define SOKUTEI_ITERATIONS 10
+#define SOKUTEI_MAX_COUNTER_COUNT  100
+#define SOKUTEI_COUNTER_LENGTH 30
+#define SOKUTEI_INTEGER_COUNTER_TYPE int
+#define SOKUTEI_FLOAT_COUNTER_TYPE double
+#define SOKUTEI_INTERVAL_TIMER_COUNTER_TYPE int
+
+#define MAX_SIZE_OF_TYPES   sizeof(SOKUTEI_INTEGER_COUNTER_TYPE) > sizeof(SOKUTEI_FLOAT_COUNTER_TYPE) \
+                            ? sizeof(SOKUTEI_INTEGER_COUNTER_TYPE) > sizeof(SOKUTEI_INTERVAL_TIMER_COUNTER_TYPE) \
+                                ? sizeof(SOKUTEI_INTEGER_COUNTER_TYPE) \
+                                : sizeof(SOKUTEI_INTERVAL_TIMER_COUNTER_TYPE)\
+                            : sizeof(SOKUTEI_FLOAT_COUNTER_TYPE) > sizeof(SOKUTEI_INTERVAL_TIMER_COUNTER_TYPE) \
+                                ? sizeof(SOKUTEI_FLOAT_COUNTER_TYPE) \
+                                : sizeof(SOKUTEI_INTERVAL_TIMER_COUNTER_TYPE)
```

Comment:
Pretty nice altogether, good job!

One minor thing: can you please just wrap the whole thing in parenthesis? Better be safe than sorry. Or, if it's safe as-is, it's okay.